src/components/Button.test.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import Button from './Button';
+import { render, screen } from '@testing-library/react';
+import fireEvent from '@testing-library/user-event';
+import { ReactComponent as IconLarge } from '../images/play-large.svg';
+
+test('check if any text is present (it should not) if you pass the iconOnly and icon props.', () => {
+  const clickPlay = jest.fn();
+  render(
+    <Button onClick={clickPlay} icon={IconLarge} iconOnly children="Play" />
+  );
+  const icon = screen.queryByTestId('buttonIcon');
+  expect(icon).toBeInTheDocument();
```

Comment:
If you expect the icon to exist, you can just use `screen.getByTestId('buttonIcon')` - it will fail if it's not found in the rendered content